test/refda/test_exec.c

```diff
@@ -406,3 +406,28 @@ void test_refda_exec_wait_cond(int delay_ms)
 
     cace_ari_deinit(&target);
 }
+
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:/TD/1, ari:/TD/60
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "820D01", "820D183C", 1, true)
+void test_refda_exec_time_based_rule(const char *actionhex, const char *starthex, const char *periodhex,
```

Comment:
Could you include a test case for both variations of the start time? That would cover the other possible starting condition.

Since the other variation is an absolute time, you could use an additional func parameter to convert the start time to absolute by adding it to the current time.

Is there a way to verify the period in the unit test if it was a shorter duration? Check for two executions, or even just check the timeline after the first execution?

Reply:
Extended the TBR unit tests to include an absolute start time, and to check for multiple iterations in order to test period.